### PR TITLE
Add com.remarkable.reMarkable.json

### PR DESCRIPTION
**Pull request by noverby**

### CODE REVIEW

File: com.remarkable.reMarkable.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    "app-id": "com.remarkable.reMarkable",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.9",
+    "sdk": "org.kde.Sdk",
+    "command": "reMarkable",
+    "finish-args": [
+        "--filesystem=home",
+        "--socket=x11",
+        "--share=network",
+        "--extra-data=remarkable-linux-client.tgz:1a1519adc6d62359f6f73bc2a4a6e76fb710fed78ca1bc09e46191e825d3058c:20777403::https://remarkable.engineering/remarkable-linux-client-0.0.5-16-1408-g7eca2b66.tgz",
```
Comment (nedrichards) [inline com.remarkable.reMarkable.json]: I assume this is an x86-64 only binary? You'll need a flathub.json file as well to limit it to that architecture as documented on the [maintainers page](https://github.com/flathub/flathub/wiki/Maintaining-an-application-on-flathub).

**Comment by nedrichards** [comment]
Test build: https://flathub.org/builds/#/builders/1/builds/998

File: com.remarkable.reMarkable.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{
+    "app-id": "com.remarkable.reMarkable",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.9",
+    "sdk": "org.kde.Sdk",
+    "command": "reMarkable",
+    "finish-args": [
+        "--filesystem=home",
+        "--socket=x11",
```
Comment (nedrichards) [inline com.remarkable.reMarkable.json]: you'll also want --share=ipc here for X11 apps.

File: com.remarkable.reMarkable.json (hunk 1 above)
Comment (noverby) [reply com.remarkable.reMarkable.json]: Fixed

File: com.remarkable.reMarkable.json (hunk 2 above)
Comment (noverby) [reply com.remarkable.reMarkable.json]: Fixed

**Comment by nedrichards** [comment]
Looks like this doesn't come with appdata. [Appdata is required for Flathub](https://github.com/flathub/flathub/wiki/Review-Guidelines) as otherwise apps don't show up in the Software store UI.

File: com.remarkable.reMarkable.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                    "dest-filename": "reMarkable",
+                    "commands": [
+                        "exec env LD_LIBRARY_PATH=/app/usr/lib TMPDIR=$XDG_CACHE_HOME /app/extra/reMarkable \"$@\""
+                    ]
+                }
+            ],
+            "modules": [
+                {
+                    "name": "qtwebsockets",
+                    "buildsystem": "simple",
```
Comment (nedrichards) [inline com.remarkable.reMarkable.json]: there's a qmake buildsystem available

Comment (noverby) [reply com.remarkable.reMarkable.json]: @nedrichards  Yeah I know, but how do you combine it with the modifications I do with sed?

Comment (nedrichards) [reply com.remarkable.reMarkable.json]: Good point, you can't do those as a post-install hook since they need to run before make

**Comment by noverby** [comment]
@nedrichards Appdata is now provided.

**Comment by nedrichards** [comment]
Test build: https://flathub.org/builds/#/builders/1/builds/1023

**Comment by noverby** [comment]
Test should pass now...

**Comment by nedrichards** [comment]
New build: https://flathub.org/builds/#/builders/1/builds/1024

**Comment by nedrichards** [comment]
OK, it'd be great if you could add screenshots to the appdata, as otherwise it'll look a bit rubbish in clients, but that's not a specific blocker

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.remarkable.reMarkable

**Pull request closed without merging**